src/core/score_map.cc

```diff
@@ -52,21 +46,33 @@ bool ScoreMap::AddOrUpdate(string_view field, double value) {
 
   absl::little_endian::Store64(newkey + meta_offset, u.u);
 
+  return newkey;
+}
+
+}  // namespace
+
+ScoreMap::~ScoreMap() {
+  Clear();
+}
+
+pair<void*, bool> ScoreMap::AddOrUpdate(string_view field, double value) {
```

Comment:
expose internal objects managed by ScoreMap so we could 
insert them into BPTree